src/compiler/moduleNameResolver.ts

```diff
@@ -1576,6 +1576,25 @@ namespace ts {
 
     /** Return the file if it exists. */
     function tryFile(fileName: string, onlyRecordFailures: boolean, state: ModuleResolutionState): string | undefined {
+        if (!state.compilerOptions.moduleSuffixes) {
+            return tryFileLookup(fileName, onlyRecordFailures, state);
+        }
+
+        const ext = tryGetExtensionFromPath(fileName) ?? "";
+        const fileNameNoExtension = ext ? removeExtension(fileName, ext) : fileName;
+
+        const suffixes = isArray(state.compilerOptions.moduleSuffixes) &&
+            state.compilerOptions.moduleSuffixes.length > 0 ? state.compilerOptions.moduleSuffixes : [""];

```

Comment:
this shouldnt be needed, our compiler options should verify that module suffixes are arrays